stdlib/email/src/main/java/org/ballerinalang/stdlib/email/server/EmailListenerHelper.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.email.server;
+
+import org.ballerinalang.jvm.BRuntime;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.stdlib.email.util.EmailConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Helper class for listener functions.
+ *
+ * @since 1.3.0
+ */
+public class EmailListenerHelper {
+
+    private EmailListenerHelper() {
+        // private constructor
+    }
+
+    /**
+     * Register a new listener for an email server endpoint.
+     * @param emailListener Listener that places emails in Ballerina runtime
+     * @param serviceEndpointConfig Email server endpoint configuration
+     * @param service Ballerina service to be listened
+     * @return Registered new Email connector with listening capability
+     */
+    public static EmailConnector register(ObjectValue emailListener, MapValue<Object, Object> serviceEndpointConfig,
+                                          ObjectValue service) {
+        EmailConnectorFactory emailConnectorFactory = new EmailConnectorFactory();
+        final EmailListener listener = new EmailListener(BRuntime.getCurrentRuntime(), service);
+        Map<String, Object> paramMap = getServerConnectorParamMap(serviceEndpointConfig);
+        EmailConnector emailConnector = emailConnectorFactory.createServerConnector(paramMap, listener);
+        emailListener.addNativeData(EmailConstants.EMAIL_SERVER_CONNECTOR, emailConnector);
+        serviceEndpointConfig.addNativeData(EmailConstants.EMAIL_SERVER_CONNECTOR, emailConnector);
+        return emailConnector;
+    }
+
+    private static Map<String, Object> getServerConnectorParamMap(MapValue serviceEndpointConfig) {
+        Map<String, Object> params = new HashMap<>(7);
+        MapValue secureSocket = serviceEndpointConfig.getMapValue(EmailConstants.ENDPOINT_CONFIG_SECURE_SOCKET);
+        if (secureSocket != null) {
+            final MapValue privateKey = secureSocket.getMapValue(EmailConstants.ENDPOINT_CONFIG_PRIVATE_KEY);
+            if (privateKey != null) {
+                final String privateKeyPath = privateKey.getStringValue(EmailConstants.ENDPOINT_CONFIG_PATH);
+                if (privateKeyPath != null && !privateKeyPath.isEmpty()) {
+                    params.put(EmailConstants.IDENTITY, privateKeyPath);
```

Comment:
Can we have a usecase where privateKeyPath is present but without privateKeyPassword?

Reply:
What if the private key is not encrypted?

Reply:
IINM, we need to have a private key right? I'm just wondering whether can we consolidate both if block to a single one.

Reply:
Consolidated.